vpr/src/draw/draw.cpp

```diff
@@ -3133,6 +3133,7 @@ static std::vector<int> trace_routed_connection_rr_nodes(const ClusterNetId net_
     std::reverse(rr_nodes_on_path.begin(), rr_nodes_on_path.end());
 
     if (allocated_route_tree_structs) {
+        free_route_tree(rt_root);
```

Comment:
It looks like rt_root is allocated regardless of the value of `allocated_route_tree_structs`. I think that means `free_route_tree()` should also be called unconditionally (i.e. not within the if body).